backend/src/business_logic_layer/map_manager.js

```diff
@@ -2,6 +2,47 @@ module.exports = function ({ mapRepository }) {
     return {
         getAllMaps: function (callback) {
             mapRepository.getAllMaps(callback)
+        },
+
+        createMap: function (callback) {
```

Comment:
Consider moving the functionality for setting and retrieving the last map Id to the map repository. Let the manager handle incrementing the id